src/runtime/manifest.ts

```diff
@@ -44,35 +45,29 @@ export class ManifestError extends Error {
 // TODO(shans): Make sure that after refactor Storage objects have a lifecycle and can be directly used
 // deflated rather than requiring this stub.
 export class StorageStub {
-  type: Type;
-  id: string;
-  name: string;
-  storageKey: string;
-  storageProviderFactory: StorageProviderFactory;
-  referenceMode = false;
-  originalId: string;
-
-  constructor(type: Type, id: string, name: string, storageKey: string, storageProviderFactory: StorageProviderFactory, originalId: string) {
-    this.type = type;
-    this.id = id;
-    this.name = name;
-    this.storageKey = storageKey;
-    this.storageProviderFactory = storageProviderFactory;
-    this.originalId = originalId;
-  }
-
-  get version(): number {
-    return undefined; // Fake to match StorageProviderBase.
-  }
-
-  get description(): string {
-    return undefined; // Fake to match StorageProviderBase;
-  }
+  constructor(public readonly type: Type, 
+              public readonly id: string,
+              public readonly name: string,
+              public readonly storageKey: string,
+              public readonly storageProviderFactory: StorageProviderFactory,
+              public readonly originalId: string,
+              public readonly description: string,
+              public readonly version?: number,
+              public readonly source?: string,
+              public readonly referenceMode: boolean = false,
+              public readonly model?: {}[]) {}
 
   async inflate() {
-    const store = await this.storageProviderFactory.connect(this.id, this.type, this.storageKey);
+    const store = await this.storageProviderFactory.connectOrConstruct(this.id, this.type, this.storageKey);
```

Comment:
can we determine whether this will be a connect or a construct from the context?

Reply:
for 'copy' and 'create' (the original use cases) we can, because these are new stores that are being created. but for 'map' we

Reply:
'map' should never be `construct` as it's mapping an existing store.

Reply:
if this 'map'-ped store is a volatile storage stub in the context, when should it be inflated?

Reply:
inflation should always happen on recipe instantiation / arc deserialization.

Reply:
We should probably discuss offline, but...
If there is a volatile storage stub in the arc's context, and it is mapped by the recipe that is being instantiated by the arc - the storage stub needs to be inflated. Will the resulting store be part of the context or the arc itself?

Also, you wrote "'map' should never be construct as it's mapping an existing store.", but for the volatile storage doesn't it need to be constructed at some point in order to come to existence?

Reply:
done - replaced `connectOrConstruct` call here with an explicit `connect` or `construct`.